### PR TITLE
Update template to require npm in non-SCL bundles

### PR DESCRIPTION
**Pull request by zjhuntin**

### CODE REVIEW

**Comment by zjhuntin** [comment]
I'm not 100% sure if this is what is needed, but I'm going to test it. Figured others might know if it'd work on sight so, made the PR.

File: bundle.mustache [hunk 1]
```diff
@@ -18,6 +18,8 @@ BuildRequires: %{?scl_prefix_nodejs}npm
 %else
 BuildRequires: nodejs-packaging
 %endif
+BuildRequires: nodejs
+BuildRequires: npm
```
Comment (ehelms) [inline bundle.mustache]: I think we'd want to move these inside the else block above so they aren't installed when built with an SCL

**Comment by zjhuntin** [comment]
http://koji.katello.org/koji/taskinfo?taskID=298770

Have it passing with this adjustment of just BuildRequires: npm

File: single.mustache [hunk 2]
```diff
@@ -15,6 +15,7 @@ URL: {{{PROJECTURL}}}
 {{/each}}
 %if 0%{?!scl:1}
 BuildRequires: nodejs-packaging
+BuildRequires: npm
```
Comment (ekohl) [inline single.mustache]: Why is NPM needed? Does that provide the packaging macros on EL8? The only reason we install it is for the `%{nodejs_sitelib}` macro and other packaging scripts. NPM itself isn't ran.

Comment (zjhuntin) [reply single.mustache]: It failed trying to call npm on the el8 builds, not sure why :-/

Comment (ehelms) [reply single.mustache]: Do you have a link to a failed task?

File: bundle.mustache (hunk 1 above)
Comment (ekohl) [reply bundle.mustache]: `if` has `%{?scl_prefix_nodejs}npm`. I guess that on EL7 `node-packaging` pulls in `npm`. Note point 3 in https://github.com/theforeman/npm2rpm/issues/17 btw.

Comment (ekohl) [reply bundle.mustache]: My suggestion would be
```
BuildRequires: %{?scl_prefix_nodejs}npm
%if 0%{!?scl:1}
BuildRequires: nodejs-packaging
%endif
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Could you have a look at my suggestion?

File: bundle.mustache [hunk 3]
```diff
@@ -13,9 +13,8 @@ URL: {{{PROJECTURL}}}
 {{#each SOURCES}}
 {{{.}}}
 {{/each}}
-%if 0%{?scl:1}
 BuildRequires: %{?scl_prefix_nodejs}npm
-%else
+%if 0%{?scl:1}
```
Comment (ehelms) [inline bundle.mustache]: I think there is a ! missing here `0%{!?scl:1}`

**Review by ekohl — APPROVED** [review]

**Pull request merged**